### PR TITLE
Add lint rules, make code conform

### PR DESCRIPTION
**Pull request by joshuaauerbachwatson**
This PR lint rules and adjusts the code to conform to them.   The intent is to enforce these rules going forward.   Initially, enforcement will be manual (`npm run lint` or configure your IDE to do the equivalent).

### CODE REVIEW

**Comment by rabbah** [comment]
Scanned all changes informing white space. LGTM. 

Noted the \\ escape in the swift sample.

**Comment by rabbah** [comment]
Scanned all changes ignoring white space. LGTM. 

Noted the \\ escape in the swift sample.

File: deployer/src/NimBaseCommand.ts [hunk 1]
```diff
@@ -62,48 +63,53 @@ export class NimFeedback implements Feedback {
...
+
+    exit(_code: number): void {
       // a no-op here
     }
 }
 
 // Test if a supplied logger is a CaptureLogger
 function isCaptureLogger(logger: NimLogger): logger is CaptureLogger {
-  return !!logger['captured']
+  return !!('captured' in logger)
```
[the remark this answers is not in the document]
Comment (joshuaauerbachwatson) [reply deployer/src/NimBaseCommand.ts:107]: Agreed.   I noticed this myself and will make the change.

**Comment by joshuaauerbachwatson** [comment]
I changed the delimiter rule for multi-line type and interface declarations to use newline instead of comma.   Also eliminated the !! as per comment from @rabbah.

**Review by bhageena — APPROVED** [review]
LGTM!

**Pull request merged**